### ISSUE TITLE
Implement an option to upload coverage data to codecov and coveralls

### ISSUE THREAD

**Issue by marco-c** [mozilla/grcov#203]
We basically need to rewrite https://github.com/mozilla/release-services/blob/eaeae5e854935dc7bce3bdcc181673f480d864a2/src/codecoverage/bot/code_coverage_bot/uploader.py in Rust.

**Comment by johnregan** [comment]
Is this ticket still relevant ? I'd like to make an attempt in order to learn some Rust.
I was unable to find uploader.py in the main branch of release-services....this code must have been extracted at some stage and this commit (eaeae5e854935dc7bce3bdcc181673f480d864a2) is all that remains.

If relevant: 

Can you provide instructions on what the desired interface should look like so my implementation could be of some use ?
In terms of testing, I found a [unit test](https://github.com/mozilla/release-services/blob/eaeae5e854935dc7bce3bdcc181673f480d864a2/src/codecoverage/bot/tests/test_uploader.py) which i was going to replicate, is there any manual test I could do to check if it's working correctly ? It would be great as I could then migrate this manual step to an integration test.

**Comment by johnregan** [comment]
@marco-c @calixteman @zommiommy @lclc

**Comment by marco-c** [comment]
@johnregan I think it'd still be nice to have this.

> Can you provide instructions on what the desired interface should look like so my implementation could be of some use ?

I don't have any explicit preference here, you can follow the same simple interface as the Python script, or implement your own thing.

> I found a unit test which i was going to replicate, is there any manual test I could do to check if it's working correctly ? It would be great as I could then migrate this manual step to an integration test.

For a manual test, I think you can create a repository, enable coveralls/codecov on it and try to upload the data.

**Comment by gsilvapt** [comment]
Should we use std lib to perform the requests or would you prefer to use std lib only?

To summarise, the idea is write a set of functions capable of:
- [ ] Uploading reports to coveralls.io;
- [ ] Uploading reports to codecov.io;
- [ ] Get latest and a specific commit data from codecov;
- [ ] Integrate with the existing CLI to allow calling the CodeCov API to upload the coverage of a given revision.

Is that it or did I get something wrong?

I would like to take this one as it sounds about a good issue for me, although I am new in Rust and in this code base - this means it will likely take me a lot of time and I might need some help occasionally. If this is a priority issue, then someone more seasoned/experienced can take it.

Thanks in advance!

**Comment by marco-c** [comment]
@gsilvapt exactly, after each step (e.g. "uploading reports to coveralls.io"), we also want to have CLI commands that use the function.

Feel free to work on this!
